worlds/blasphemous/Rules.py

```diff
@@ -1,4458 +1,1373 @@
+            "guiltRooms5":            guilt_indirect_regions,
+            "guiltRooms6":            guilt_indirect_regions,
+            "guiltRooms7":            guilt_indirect_regions,
+            "swordRooms1":            sword_indirect_regions,
+            "swordRooms2":            sword_indirect_regions,
+            "swordRooms3":            sword_indirect_regions,
+            "swordRooms4":            sword_indirect_regions,
+            "swordRooms5":            sword_indirect_regions,
+            "swordRooms6":            sword_indirect_regions,
+            "swordRooms7":            sword_indirect_regions,
+            "redentoRooms2":          redento_indirect_regions,
+            "redentoRooms3":          redento_indirect_regions,
+            "redentoRooms4":          redento_indirect_regions,
+            "redentoRooms5":          redento_indirect_regions,
+            "miriamRooms5":           miriam_indirect_regions,
+            "chaliceRooms3":          chalice_indirect_regions
+        }
 
-def charge_beam(state: CollectionState, player: int) -> bool:
-    return state.has("Charged Skill", player, 3)
+        self.indirect_regions["amanecidaRooms1"] = [*self.indirect_regions["canBeatGraveyardBoss"],
+                                                    *self.indirect_regions["canBeatJondoBoss"],
+                                                    *self.indirect_regions["canBeatPatioBoss"],
+                                                    *self.indirect_regions["canBeatWallBoss"]]
+        self.indirect_regions["amanecidaRooms2"] = [*self.indirect_regions["canBeatGraveyardBoss"],
+                                                    *self.indirect_regions["canBeatJondoBoss"],
+                                                    *self.indirect_regions["canBeatPatioBoss"],
+                                                    *self.indirect_regions["canBeatWallBoss"]]
+        self.indirect_regions["amanecidaRooms3"] = [*self.indirect_regions["canBeatGraveyardBoss"],
+                                                    *self.indirect_regions["canBeatJondoBoss"],
+                                                    *self.indirect_regions["canBeatPatioBoss"],
+                                                    *self.indirect_regions["canBeatWallBoss"]]
+        self.indirect_regions["amanecidaRooms4"] = [*self.indirect_regions["canBeatGraveyardBoss"],
+                                                    *self.indirect_regions["canBeatJondoBoss"],
+                                                    *self.indirect_regions["canBeatPatioBoss"],
+                                                    *self.indirect_regions["canBeatWallBoss"]]
+
+
+    def req_is_region(self, string: str) -> bool:
+        if (string[0] == "D" and string[3] == "Z" and string[6] == "S")\
+        or (string[0] == "D" and string[3] == "B" and string[4] == "Z" and string[7] == "S"):
+            return True
+        return False
```

Comment:
Could just do this ?
```suggestion
        return ((string[0] == "D" and string[3] == "Z" and string[6] == "S")
                or (string[0] == "D" and string[3] == "B" and string[4] == "Z" and string[7] == "S"))
```